### ISSUE TITLE
It’s difficult to apply the same styles for the element for `focused` and `opened` states

### ISSUE THREAD

**Issue by jouni** [vaadin/vaadin-dropdown-menu#54]
As the internal vaadin-text-field has the focused styles (after #53 is possibly fixed), it would be great to use the same styles also when the dropdown is open, as the field is still semantically focused. 

It is the same with vaadin-combo-box – you have the dropdown visible and can navigate the list with the arrow keys while the main host element is still focused.

**Comment by jouni** [comment]
Keeping the `focused` state applied while the overlay is open would also prevent the “focus flash” on the text-field when the overlay is closed.

**Comment by web-padawan** [comment]
@jouni so we should keep `focused` attribute on both host element and internal `vaadin-text-field` like `vaadin-combo-box does`, right?

**Comment by jouni** [comment]
Yeah, I think so (sorry for the late reply).

**Comment by jouni** [comment]
There seems to be a problem with this when the `[phone]` mode is applied. When I close the overlay, the focused state is still applied, and can’t be removed by tapping outside the element. Can be reproduced with the Chrome device emulator.

Also, **one addition to the specs** (didn’t consider this before):
- When the overlay is closed by clicking outside the overlay (i.e. not selecting an item), the field should be blurred
- If the overlay is closed by selecting an item or by pressing ESC, the field should stay focused

Should I create a new issue for these?
